File: tests_wraps/src/main/java/com/github/klee0kai/test/car/di/inject/CarInjectComponent.java
package com.github.klee0kai.test.car.di.inject;

import com.github.klee0kai.stone.annotations.component.Component;
import com.github.klee0kai.stone.annotations.component.ProtectInjected;
import com.github.klee0kai.test.car.di.wrapped.custom.wrappers.CarProviderWrapper;
import com.github.klee0kai.test.car.model.CarInject;
import com.github.klee0kai.test.car.model.CarInjectLists;
import com.github.klee0kai.test.car.model.CarInjectProvider;

@Component(
        wrapperProviders = {CarProviderWrapper.class}
)
public abstract class CarInjectComponent {

    protected abstract CarInjectModule module();

    public abstract void inject(CarInject carInject);

    public abstract void inject(CarInjectLists carInject);

    public abstract void inject(CarInjectProvider carInject);

    @ProtectInjected
    public abstract void protect(CarInject carInject);

    @ProtectInjected
    public abstract void protect(CarInjectLists carInject);

    @ProtectInjected
    public abstract void protect(CarInjectProvider carInject);

}

File: tests_wraps_kotlin/src/main/kotlin/com/github/klee0kai/test/car/di/inject/CarInjectComponent.kt
package com.github.klee0kai.test.car.di.inject

import com.github.klee0kai.stone.annotations.component.Component
import com.github.klee0kai.stone.annotations.component.ProtectInjected
import com.github.klee0kai.test.boxed.di.inject.CarBoxedWrapper
import com.github.klee0kai.test.car.model.CarInject
import com.github.klee0kai.test.car.model.CarInjectLists
import com.github.klee0kai.test.car.model.CarInjectProvider

@Component(wrapperHelpers = [CarBoxedWrapper::class])
abstract class CarInjectComponent {

    abstract fun module(): CarInjectModule?
    abstract fun inject(carInject: CarInject?)
    abstract fun inject(carInject: CarInjectLists?)
    abstract fun inject(carInject: CarInjectProvider?)

    @ProtectInjected
    abstract fun protect(carInject: CarInject?)

    @ProtectInjected
    abstract fun protect(carInject: CarInjectLists?)

    @ProtectInjected
    abstract fun protect(carInject: CarInjectProvider?)
}
